lib/matplotlib/scale.py

```diff
@@ -82,6 +82,97 @@ def get_transform(self):
         return IdentityTransform()
 
 
+class FuncTransform(Transform):
+     """
+     A simple transform that takes and arbitrary function for the
+     forward and inverse transform.
+     """
+
+     input_dims = 1
+     output_dims = 1
+     is_separable = True
+     has_inverse = True
+
+     def __init__(self, forward, inverse):
+         """
+         Parameters
+         ----------
+
+         forward: callable
+             The forward function for the transform.  This function must have
+             an inverse and, for best behavior, be monotonic.
+             It must have the signature::
+
+                def forward(values: array-like) ->
+                        array-likeThe forward function for the transform
+
+         inverse: callable
+            The inverse of the forward function.  Signature as ``forward``.
+         """
+         super().__init__()
+         if callable(forward) and callable(inverse):
+             self._forward = forward
+             self._inverse = inverse
+         else:
+             raise ValueError('arguments to FuncTransform must '
+                              'be functions')
+
+     def transform_non_affine(self, values):
+         return self._forward(values)
+
+     def inverted(self):
+         return FuncTransform(self._inverse, self._forward)
+
+
+class FuncScale(ScaleBase):
+     """
+     Provide an arbitrary scale with user-supplied function for the axis.
+     """
+
+     name = 'function'
+
+     def __init__(self, axis, functions):
+         """
+         Parameters
+         ----------
+
+         axis: the axis for the scale
+
+         functions: (callable, callable)
+            two-tuple of the forward and inverse functions for the scale.
+            The forward function must have an inverse and, for best behavior,
+            be monotonic.
+
+            Both functions must have the signature::
+
+               def forward(values: array-like) ->
+                       array-likeThe forward function for the transform
+         """
+         forward, inverse = functions
+         transform = FuncTransform(forward, inverse)
+         self._transform = transform
+
+     def get_transform(self):
+         """
+         The transform for arbitrary scaling
+         """
+         return self._transform
+
+     def set_default_locators_and_formatters(self, axis):
+         """
+         Set the locators and formatters to the same defaults as the
+         linear scale.
+         """
+         axis.set_major_locator(AutoLocator())
+         axis.set_major_formatter(ScalarFormatter())
+         axis.set_minor_formatter(NullFormatter())
+         # update the minor locator for x and y axis based on rcParams
+         if rcParams['xtick.minor.visible']:
+             axis.set_minor_locator(AutoMinorLocator())
```

Comment:
Can you incoorporate the fix from #12938 here as well?

Reply:
Sorry for the delay - Done!